### PR TITLE
Update libprotobuf with conda-forge changes to v3.19.1

### PR DESCRIPTION
**Pull request by ScottBailey**
Check upstream:
   https://github.com/protocolbuffers/protobuf/tree/v3.19.1

Check pinnings:
   https://github.com/protocolbuffers/protobuf/blob/v3.19.1/protobuf_version.bzl
   https://github.com/protocolbuffers/protobuf/blob/v3.19.1/protobuf_deps.bzl
   https://github.com/protocolbuffers/protobuf/blob/v3.19.1/post_process_dist.sh
   https://github.com/protocolbuffers/protobuf/blob/v3.19.1/appveyor.bat

Review Changelog:
   https://github.com/protocolbuffers/protobuf/blob/v3.19.1/CHANGES.txt
   Mostly bugfixes, with the following change that may affect some users:
      * Protobuf python generated code are simplified. Some platforms that uses
         "is"("is not") to compare the enum or descriptor's label/type may fail,
          should use "=="("!=") instead.

Verify dev url, doc url. 

Ran tests:
   `conda build ../wip/libprotobuf-feedstock/ --test`
   resulting in `All tests passed`.

In addition the package built on concourse for the following architectures: win64, lin64, ppc64le, osx.

In conclusion I feel it's safe to update libprotobuf to v3.19.1

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
the PR is comming from the right place
the package is relatively popular with 8 million downloads
the change is a minor version change
there is a minor issue to keep track of, on protobuf:
  https://github.com/conda-forge/libprotobuf-feedstock/issues/68

`patch` and `sed` were added to the requirements, `s390x` was added as an architecture.
the location of `sed` was modified on the `build-share.sh` file
The package was released in october 28 2021

There is a known open issue with `cmake` on `libprotobuf` for some users

**Review by varlackc — APPROVED** [review]

**Review by antkneed — APPROVED** [review]
LGTM 👍

**Review by skupr-anaconda — APPROVED** [review]

File: .ci_support/linux_64_.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ c_compiler_version:
 cdt_name:
 - cos6
 channel_sources:
-- conda-forge,defaults
+- conda-forge
```
Comment (katietz) [inline .ci_support/linux_64_.yaml:8]: hmm, we shouldn't have references here to conda-forge.  so defaults is the choice we want

File: .ci_support/linux_64_.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ c_compiler_version:
...
 channel_targets:
```
Comment (katietz) [inline .ci_support/linux_64_.yaml:9]: same here ... channel_targets are for sure not conda-forge

File: .ci_support/linux_ppc64le_.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ c_compiler_version:
 cdt_name:
 - cos7
 channel_sources:
-- conda-forge,defaults
+- conda-forge
```
Comment (katietz) [inline .ci_support/linux_ppc64le_.yaml:8]: likewise ... but as those are items of CF's built, it shouldn't matter too much

File: recipe/build-shared.sh [hunk 4]
```diff
@@ -13,6 +13,7 @@ then
     # to improve performance, disable checks intended for debugging
     CXXFLAGS="$CXXFLAGS -DNDEBUG"
 elif [ "$(uname)" == "Darwin" ];
+    export SED="${BUILD_PREFIX}/bin/sed"
 then
```
Comment (katietz) [inline recipe/build-shared.sh]: this looks like a typo.  then after a statement?

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
   version: {{ version }}
 
 source:
   - url: https://github.com/protocolbuffers/protobuf/archive/v{{ version }}/libprotobuf-v{{ version }}.tar.gz
-    sha256: 36f81e03a0702f8f935fffd5a486dac1c0fc6d4bae1cd02c7a32448ad6e63bcb
+    sha256: 87407cd28e7a9c95d9f61a098a53cf031109d451a7763e7dd1253abf8b4df422
     patches:
       - 0001-remove-Werror-from-test-flags.patch  # [ppc64le or aarch64 or s390x]
-      - 0002-remove-failing-json-parser-test.patch  # [s390x]
+      # - 0002-remove-failing-json-parser-test.patch  # [s390x]
```
Comment (katietz) [inline recipe/meta.yaml:12]: hmm, those patches might be worth being ported?  Not sure if for our builders memory is big enough for those tests on s390x ...

File: recipe/meta.yaml [hunk 6]
```diff
@@ -26,13 +26,20 @@ source:
     folder: third_party/googletest
 
 build:
-  number: 1
+  number: 0
+
+requirements:
+  build:
+    - patch  # [not win]
```
Comment (katietz) [inline recipe/meta.yaml]: I guess that here m2-patch fow Windows is missing.  As this recipe seems to work on Windows too, as there is no skip for it

**Review by katietz — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 6 above)
Comment (katietz) [reply recipe/meta.yaml]: hmm, maybe not ... those patchees are not applied for Windows ...  So add to the 'patches:' line the jinja2 preprocessor statement '[ppc64le or aarch64 or s390x]'. So we avoid and empty yaml entry

File: .ci_support/linux_64_.yaml (hunk 1 above)
Comment (ScottBailey) [reply .ci_support/linux_64_.yaml:8]: This is a merge from conda-forge and we don't use `.ci_support` so it should be an acceptable change. We've done the merge to make future merges easier.

File: .ci_support/linux_64_.yaml (hunk 2 above)
Comment (ScottBailey) [reply .ci_support/linux_64_.yaml:9]: This is a merge from conda-forge and we don't use `.ci_support` so it should be an acceptable change. We've done the merge to make future merges easier.

File: recipe/build-shared.sh (hunk 4 above)
Comment (ScottBailey) [reply recipe/build-shared.sh]: Moved into it's own block.

File: recipe/meta.yaml (hunk 5 above)
Comment (ScottBailey) [reply recipe/meta.yaml:12]: See next comment...

File: recipe/meta.yaml (hunk 6 above)
Comment (ScottBailey) [reply recipe/meta.yaml]: Made entries more specific.

File: recipe/meta.yaml [hunk 7]
```diff
@@ -50,6 +57,7 @@ outputs:
         - pkg-config  # [not win]
         - unzip  # [not win]
         - make  # [not win]
+        - sed  # [osx]
```
Comment (antkneed) [inline recipe/meta.yaml]: Since this is only needed on the **osx-arm64** worker, we can be more explicit here.
```suggestion
        - sed  # [osx and arm64]
```

File: recipe/meta.yaml [hunk 8]
```diff
@@ -78,6 +86,7 @@ outputs:
...
```
Comment (antkneed) [inline recipe/meta.yaml]: If only the `build-shared.sh` has the `sed` workaround, is this needed for `libprotobuf-static`?

**Review by antkneed — APPROVED** [review]
Two straight-forward, minor comments.

File: recipe/meta.yaml (hunk 8 above)
Comment (ScottBailey) [reply recipe/meta.yaml]: `sed` workaround is unnecessary for static builds, removed.

File: recipe/meta.yaml (hunk 7 above)
Comment (ScottBailey) [reply recipe/meta.yaml]: Fixed.

**Review by katietz — APPROVED** [review]
thanks!
